### PR TITLE
Use serde for JSON and struct deserialization.

### PR DESCRIPTION
**Pull request by jimmycuadra**
Addresses #63. If this is accepted I will follow up by switching [params](https://github.com/iron/params) over to Serde as well.

### CODE REVIEW

**Comment by reem** [comment]
Travis is failing since the examples have not been updated.

**Comment by reem** [comment]
Otherwise this looks good, but I'm interested to see how the examples change before merging (also I'd like to keep travis green).

**Comment by jimmycuadra** [comment]
Oops! Sorry for overlooking that. I wish GitHub would create a notification when one of my PRs fails a build check. Pushing up the updated example now.

**Comment by jimmycuadra** [comment]
Anything else to do here?

**Comment by jimmycuadra** [comment]
I discovered that if a build script introduces syntex into the crate's dependencies, any upstream crates that use bodyparser won't work on nightly Rust (quasi_macros fails to compile because it gets confused between syntex and rustc_private). There's no way I'm aware of to restrict the build script/build dependencies to the examples, so I opted to just remove the executable examples. Not ideal, but it's still in the README for people to get an idea of how bodyparser works.

**Comment by jimmycuadra** [comment]
I just didn't feel right about removing the executable example, so I implemented Serde deserialization manually. It's a lot of extra code that obscures the example somewhat, but at least you can actually try it out on stable or nightly without a build script. The version with the automatic derivation is still in the README where it's more important to be concise and to the point.

File: examples/bodyparser.rs [hunk 1]
```diff
@@ -1,18 +1,97 @@
...
+use serde::de::{MapVisitor, Visitor};
 
-#[derive(Debug, Clone, RustcDecodable)]
+#[derive(Debug, Clone)]
 struct MyStructure {
     a: String,
     b: Option<String>,
 }
 
+impl Deserialize for MyStructure {
```
Comment (reem) [inline examples/bodyparser.rs]: Can you add a comment here explaining why we are doing this and maybe linking to how not to have to do this?

**Pull request merged**